plugins/utilities/desktop_rpc.py

```diff
@@ -0,0 +1,464 @@
+# Released under the MIT License. See LICENSE for details.
+#
+"""placeholder :clown:"""
+
+# ba_meta require api 7
+
+from __future__ import annotations
+from typing import TYPE_CHECKING
+from codecs import encode
+from pypresence.utils import get_event_loop
+import _ba
+import ba
+import threading
+import time
+import pypresence
+import json
+import uuid
+import ast
+import asyncio
+from urllib.request import Request, urlopen, urlretrieve
+from pathlib import Path
+
+# installing pypresence
+
+
+def get_module():
+    import os
+    import zipfile
+    install_path = Path(f"{os.getcwd()}/ba_data/python")
+    path = Path(f"{install_path}/pypresence.zip")
+    if not os.path.exists(Path(f"{install_path}/pypresence")):
+        url = "https://github.com/brostosjoined/BombsquadRPC/releases/download/presence-1.0/pypresence.zip"
+        filename, headers = urlretrieve(url, filename=path)
+        with zipfile.ZipFile(path) as f:
+            f.extractall(install_path)
+        os.remove(path)
+
+
+get_module()
+
+
+if TYPE_CHECKING:
+    from typing import Any, Tuple
+
+DEBUG = True
+
+
+def print_error(err: str, include_exception: bool = False) -> None:
+    if DEBUG:
+        if include_exception:
+            ba.print_exception(err)
+        else:
+            ba.print_error(err)
+    else:
+        print(f"ERROR in discordrp.py: {err}")
+
+
+def log(msg: str) -> None:
+    if DEBUG:
+        print(f"LOG in discordrp.py: {msg}")
+
+
+def _run_overrides() -> None:
+    old_init = ba.Activity.__init__
+
+    def new_init(self, *args: Any, **kwargs: Any) -> None:  # type: ignore
+        old_init(self, *args, **kwargs)
+        self._discordrp_start_time = time.mktime(time.localtime())
+
+    ba.Activity.__init__ = new_init  # type: ignore
+
+    old_connect = _ba.connect_to_party
+
+    def new_connect(*args, **kwargs) -> None:  # type: ignore
+        global _last_server_addr
+        global _last_server_port
+        old_connect(*args, **kwargs)
+        c = kwargs.get("address") or args[0]
+        _last_server_port = kwargs.get("port") or args[1]
+
+    _ba.connect_to_party = ba.internal.connect_to_party = new_connect
+
+
+start_time = time.time()
+
+
+class RpcThread(threading.Thread):
+    def __init__(self):
+        super().__init__()
+        self.rpc = pypresence.Presence(963434684669382696)
```

Comment:
let's also offer a way (plugin ui settings?) to let users feed in their own app client client id

Reply:
Thats the Bombsquad client id created from [discord] (https://discord.com/developers/applications) if it changes the assets would break.

Reply:
Thats the Bombsquad client id created from [discord] (https://discord.com/developers/applications) if it changes the assets would break.